src/passes/StackCheck.cpp

```diff
@@ -127,9 +127,9 @@ struct EnforceStackLimits : public WalkerPass<PostWalker<EnforceStackLimits>> {
   }
 
 private:
-  Global* stackPointer;
-  Global* stackBase;
-  Global* stackLimit;
+  const Global* stackPointer;
+  const std::unique_ptr<Global>& stackBase;
+  const std::unique_ptr<Global>& stackLimit;
```

Comment:
Why are these different from `stackPointer`?

Reply:
`stackPointer` is initialized via `getStackPointerGlobal` in wasm-emscripten.cpp. I looked into making that function return `const std::unique_ptr<Global>&` rather than `Global*`, but ran into the problem the function returns `nullptr` on failure and there's no clean way to do that using references. With C++17, I could use `std::optional<std::reference_wrapper<Global>>`, but even that's so gross that I would rather just use `Global*` 😅

Reply:
What I was asking is about `stackBase` and `stackLimit`, that why can't they also just be `const Global*`, by passing the raw pointer using `get()`, when we call `EnforceStackLimits` constructor?

Reply:
Gentle ping :) This is a pretty minor question, but just in case you haven't had a chance to see it.